src/server/router/chat.ts

```diff
@@ -40,6 +41,76 @@ async function executeScript(
   });
 }
 
+async function updateContentDB(
+  prisma: PrismaClient,
+  chatId: string,
+  conversationHistory: any[],
+  message: string
+): Promise<Chat> {
+  conversationHistory.push({
+    sender: "UrbanGPT",
+    text: message,
+  });
+
+  const updatedContent = JSON.stringify(conversationHistory);
+  const updatedChat = await prisma.chat.update({
+    where: {
+      id: chatId,
+    },
+    data: {
+      content: updatedContent,
+      updatedAt: new Date(),
+    },
+  });
+  return updatedChat;
+}
+
+async function validateConversation(
+  prisma: PrismaClient,
+  chatId: string,
+  conversationHistory: any[],
+  message: string
+) {
+  console.log("chatid", chatId);
+  // In case the conversation has been updated in the DB through another browser
+  const chat = await prisma.chat.findUnique({
+    where: {
+      id: chatId,
+    },
+    select: {
+      content: true,
+    },
+  });
+
+  if (!chat) {
+    throw new Error("Chat not found");
+  }
+
+  // Parse the content from the database
+  let currentDbContent;
+  try {
+    currentDbContent = JSON.parse(chat.content);
+  } catch (error) {
+    console.error("Error parsing JSON from DB:", error);
+    throw new Error("Failed to parse chat content from the database");
+  }
+
+  // Compare the conversation history with DB content
+  let clientHistoryWithoutLastMessage = conversationHistory.slice(0, -1);
+  if (currentDbContent.length !== clientHistoryWithoutLastMessage.length) {
```

Comment:
Comparing length to find the latest chat message is a compromise given just because the browser message is longer it doesn't mean it should be hold as the valid one